lib/schedule.js

```diff
@@ -0,0 +1,224 @@
+
+const CRON_JOBS = {};
+
+async function initScheduledJobs() {
+  try {
+    const scheduledJobsList = await getScheduledJobs();
+
+    if (scheduledJobsList.length == 0) {
+      console.log("No scheduled-jobs found that need to be added to cron.");
+      return;
+    }
+
+    await addScheduledJobsToCron(scheduledJobsList);
+    console.log(`${scheduledJobsList.length} scheduled-jobs have been added to cron.`);
+  } catch (err) {
+    console.error(`Something went wrong while initializing scheduled-jobs`);
+    console.error(err);
+  }
+}
+
+async function getScheduledJobs() {
+  try {
+    const scheduledJobsQuery = `
+      ${PREFIXES}
+      SELECT DISTINCT ?uri ?frequency WHERE {
+        ?uri a cogs:ScheduledJob;
+              mu:uuid ?uuid;
+              task:schedule ?schedule.
+
+        ?schedule schema:repeatFrequency ?frequency.
+      }
+    `;
+
+    const scheduledJobsList = parseResult(await query(scheduledJobsQuery));
+    return scheduledJobsList;
+  } catch(err) {
+    console.error(`Something went wrong while querying scheduled-jobs`);
+    console.error(err);
+  }
+}
+
+async function addScheduledJobsToCron(scheduledJobs) {
+  scheduledJobs.forEach(job => {
+    CRON_JOBS[job.uri] = new CronJob(job.frequency, function() {
+      createNewJob(job);
+    }, null, true);
+  });
+}
+
+async function createNewJob(scheduledJob) {
+  try {
+    const scheduledJobData = await getScheduledJobData(scheduledJob);
+    const newJobQuery = await constructNewJobQuery(scheduledJobData);
+
+    await update(newJobQuery);
+  } catch (err) {
+    console.error(`Something went wrong while creating new job from scheduled-job '${scheduledJob.uri}'`);
+    console.error(err);
+  }
+}
+
+async function getScheduledJobData(scheduledJob) {
+  try {
+    const scheduledJobDataQuery = `
+      ${PREFIXES}
+      SELECT DISTINCT ?graph ?uri ?operation ?sourceUrl ?taskOperation WHERE {
+          GRAPH ?graph {
+          ?uri a cogs:ScheduledJob;
+                        task:operation ?operation.
+
+          ?scheduledTask dct:isPartOf ?uri;
+                        task:inputContainer ?inputContainer;
+                        task:operation ?taskOperation.
+    
+          ?inputContainer task:hasHarvestingCollection/dct:hasPart/nie:url ?sourceUrl.
+          
+          BIND(${sparqlEscapeUri(scheduledJob.uri)} as ?uri) 
+        }
+      }
+    `;
+
+    const scheduledJobData = parseResult(await query(scheduledJobDataQuery))[0];
+    return scheduledJobData;
+  } catch(err) {
+    console.error(`Something went wrong while querying 'graph', 'operation' & 'sourceUrl' of ${scheduledJob.uri}`);
+    console.error(err);
+  }
+}
+
+async function constructNewJobQuery(scheduledJob) {
+  try {
+    const jobId = uuid();
+    const jobUri = `http://redpencil.data.gift/id/job/${jobId}`;
+    const remoteDataObjectId = uuid();
+    const remoteDataObjectUri = `http://data.lblod.info/id/remote-data-objects/${remoteDataObjectId}`;
```

Comment:
Partially ignore what I said. You're right, this code is required if we allow users to edit the contents of data container. This has far more deeper repercussions than I had anticipated it would have. And would want to avoid these. The jobs-controller should be agnostic about the contents of the data container. 
And I don't see a simple way right now to combine EDIT of container contents and a jobs-controller that can deep-copy its contents without knowing it (in code) what the contents would be. A way out of this situation, would e.g. a configurable shacl profile,  that the code can use as guide to see what needs to be deep-copied. But this requires further thinking, and will be a bridge to far now. 

So where this this leads to. I guess, simply put: not allowing the user to edit a job (except maybe the title)